File: 402. Remove K Digits/ac.cpp
#include<string>
using namespace std;


class Solution {
public:
    string removeKdigits(string num, int k) {
        while(k > 0){
            int i = 0;
            int j = 1;
            while(num[i] <= num[i + 1] && i + 1 < num.size())              
                i++;  
            if(num[i + 1] == '0' && i == 0){
                j = i + 1;
                while(num[j] == '0')
                    j++;
                num.erase(i, j - i);
            }else
                num.erase(i, 1);
            k--;   
        }
        return num == "" ? "0" : num;
    }
};

int main(){
    Solution s;
    string a = "112";
    s.removeKdigits(a, 2);
}

File: 136.只出现一次的数字 ⭐ 位操作/ac.py
class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        a = 0
        for num in nums:
            a = a ^ num
        return a


s = Solution()
s.singleNumber([8, 2, 3, 2, 3])


'''
这个题用异或操作可以巧妙的解决，任意数异或0都等于0，且任何数异或同一个数两次会变成原来的数。
异或还可以用来交换A B 
A = A ^ B  
B = A ^ B
A = A ^ B
'''